### PR TITLE
Fix CI on 0-10-stable

### PR DESCRIPTION
**Pull request by bf4**
#### Purpose

Fix CI

I'll accept PRs into the fix_ci branch if anyone wants to help

#### Changes

A bunch because it's been broken for a while.

Each change is isolated in a commit, though, so should be easy enough to follow

#### Additional helpful information

e.g. in 

https://github.com/rails-api/active_model_serializers/pull/2179 

- [Travis Failures](https://travis-ci.org/rails-api/active_model_serializers/builds/269424966)
- [Appveyor Failures](https://ci.appveyor.com/project/bf4/active-model-serializers/build/1.0.1032-0-10-stable)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -37,18 +38,12 @@ env:
...
-  - rvm: jruby-9.1.5.0
-    env: RAILS_VERSION=5.0
-  - rvm: jruby-head
-    env: RAILS_VERSION=5.0
+  - { rvm: 2.1,           env: RAILS_VERSION=master }
+  - { rvm: jruby-9.1.5.0, env: RAILS_VERSION=master }
+  - { rvm: jruby-head,    env: RAILS_VERSION=master }
+  - { rvm: 2.1,           env: RAILS_VERSION=5.0 }
+  - { rvm: jruby-9.1.5.0, env: RAILS_VERSION=5.0 }
+  - { rvm: jruby-head,    env: RAILS_VERSION=5.0 }
```
Comment (bf4) [inline .travis.yml]: this is just for readability

File: Gemfile [hunk 2]
```diff
@@ -23,13 +24,18 @@ if version == 'master'
...
   gem 'railties', gem_version
   gem 'activesupport', gem_version
   gem 'activemodel', gem_version
   gem 'actionpack', gem_version
   gem 'activerecord', gem_version, group: :test
 end
 
+if RUBY_VERSION < '2.1'
+  gem 'nokogiri', '< 1.7'
+end
```
Comment (bf4) [inline Gemfile]: ✂️ we don't test against lower than ruby 2.1

File: Gemfile [hunk 3]
```diff
@@ -47,7 +53,7 @@ group :test do
   gem 'codeclimate-test-reporter', require: false
   gem 'm', '~> 1.5'
   gem 'pry', '~> 0.10'
-  gem 'pry-byebug', '~> 3.4', platform: :ruby
+  gem 'pry-byebug', '>= 3.4', platform: :ruby
```
Comment (bf4) [inline Gemfile]:
```
Fetching gem metadata from https://rubygems.org/...........
Fetching version metadata from https://rubygems.org/..
Fetching dependency metadata from https://rubygems.org/.
Resolving dependencies...
byebug-9.1.0 requires ruby version >= 2.2.0, which is incompatible with the
current version, ruby 2.1.5p273
```

**Comment by NullVoxPopuli** [comment]
LGTM!

**Pull request merged**